source/core_json.c

```diff
@@ -1138,6 +1167,55 @@ JSONStatus_t JSON_Validate( const char * buf,
 
 /** @cond DO_NOT_DOCUMENT */
 
+/**
+ * @brief Output index and length for the next value.
+ *
+ * Also advances the buffer index beyond the value.
+ * The value may be a scalar or a collection.
+ *
+ * @param[in] buf  The buffer to parse.
+ * @param[in,out] start  The index at which to begin.
+ * @param[in] max  The size of the buffer.
+ * @param[out] value  A pointer to receive the index of the value.
+ * @param[out] valueLength  A pointer to receive the length of the value.
+ *
+ * @return true if a value was present;
+ * false otherwise.
+ */
+static bool_ nextValue( const char * buf,
+                        size_t * start,
+                        size_t max,
+                        size_t * value,
+                        size_t * valueLength )
+{
+    bool_ ret = true;
+    size_t i, valueStart;
+
+    assert( ( buf != NULL ) && ( start != NULL ) && ( max > 0U ) );
+    assert( ( value != NULL ) && ( valueLength != NULL ) );
+
+    i = *start;
+    valueStart = i;
+
+    if( ( skipAnyScalar( buf, &i, max ) == true ) ||
+        ( skipCollection( buf, &i, max ) == JSONSuccess ) )
+    {
+        *value = valueStart;
+        *valueLength = i - valueStart;
```

Comment:
Does this mean that the input `*start` value should already be representing the next value's index?

Reply:
yes

Reply:
It would be good to mention that as a _requirement_ in the function documentation.

Reply:
added a comment